File: 2418-sort-the-people/2418-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        HashMap<Integer, String> map = new HashMap<>();
        String[] str = new String[names.length];

        for(int i = 0; i < names.length; i++)
            map.put(heights[i], names[i]);
        
        Integer[] sortedHeights = map.keySet().toArray(new Integer[0]);
        Arrays.sort(sortedHeights);
        int j = 0;
        for(int i = sortedHeights.length-1; i >= 0; i--)
            str[j++] = map.get(sortedHeights[i]);
        
        return str;
    }
}

File: 2418-sort-the-people/2418-sort-the-people.swift
class Solution {
    func sortPeople(_ names: [String], _ heights: [Int]) -> [String] {
        var res = [String]()
        var heights = heights
        
        var i = 0
        var dictionary = [Int : String]()    // heights are unique, names are not!
        
        while i < names.count {
            dictionary[heights[i]] = names[i]
            i += 1
        }
        
        heights = heights.sorted(by: >)
        
        for i in heights {
            for (j, val) in dictionary {    // j: Int, val: String
                if i == j {
                    res.append(val)
                    break
                }
            }
        }
        
        return res
    }
}